docs/document-qa.md

```diff
@@ -0,0 +1,117 @@
+
+When a `DocumentCorpus` is provided, FewShots are optional (hence the empty array that we passed in). If you need FewShots, see [Using Fewshots with Docs](#using-fewshots-with-docs).
+
+Once ready, you can deploy your agent using `fixie deploy`. Note that depending on the size and number of documents, indexing can take some time. While indexing is in progress, your agent will deploy but will always return, `I'm still starting up. Please try again in a few minutes.` See [Monitoring Indexing Status](#monitoring-indexing-status) for more.
+
+## Accessing Private URLs
+
+In many cases, the set of documents that you want to access will be behind authentication. Fixie supports passing in a `Bearer` token with each request.
+
+To enable this, you'll define a new function that returns a token that will be passed in to the `Authorization` HTTP header as follows: `Authorization: Bearer {your_token}`.
+
+Here's an example:
+
+```python
+import fixieai
+
+BASE_PROMPT = """I can answer questions from secret, non-public docs."""
+
+URLS = [
+    "https://private.myamazingsite.com/*"
+]
+
+CORPORA = [fixieai.DocumentCorpus(urls=URLS, auth_token_func="auth")]
+agent = fixieai.CodeShotAgent(BASE_PROMPT, [], CORPORA, conversational=True)
+
+@agent.register_func
+def auth():
+    return "12345"
+```
+
+## Using Fewshots with Docs
+
+By default, you don't need to use FewShots with Docs. Fixie's automatic behavior will produce a Q/A agent that meets the general need.
+
+If using FewShots, you'll need to manually tell Fixie to query the corpus using the built-in `fixie_query_corpus` function.
+
+Here's an example for an Agent that has access to docs about the primary plot points from HBO's Silicon Valley:
+
+```python
+FEW_SHOTS = """
+Q: Who was Gilfoyle played by?
+Ask Func[fixie_query_corpus]: Who was Gilfoyle played by?
+Func[fixie_query_corpus] says: Gilfoyle was played by Martin Starr.
+A: Gilfoyle was played by Martin Starr.
+"""
+```
+
+## Excluding documents during crawl
+
+There are some cases where you don't want Fixie to crawl certain wepages or types of webpages when using the `*` wildcard pattern. In order to do this, you can supply a set of `exclude_patterns`, which are an array of regular expressions:
+
+```python
+import fixieai
+
+BASE_PROMPT = """I answer questions based on the supplied corpus"""
+
+URLS = [
+    "https://public.myamazingsite.com/*"
+]
+
+# Don't index any PDFs
+EXCLUDE_PATTERNS = [
+  "*.pdf",
+]
+
+CORPORA = [fixieai.DocumentCorpus(urls=URLS, exclude_patterns=EXCLUDE_PATTERNS)]
+agent = fixieai.CodeShotAgent(BASE_PROMPT, [], CORPORA, conversational=True)
+```
+
+## Supported file types
+
+| File Type | Extension                        |
+| --------- | -------------------------------- |
+| Documents | `.doc`, `.docx`, `.ppt`, `.pptx` |
```

Comment:
Thanks for the docs!

Not sure if this is related, but when I tried to load `.doc` files, I got an error saying I need to install `libreoffice`. For mac it was through `brew install --cask libreoffice`, but I'm not sure if this works on the server already or not. Just thought to point it out.